astropy/stats/spatial.py

```diff
@@ -0,0 +1,314 @@
+        if isinstance(value, (float, int)) and value > 0:
+            self._area = value
+        else:
+            raise ValueError('area is expected to be a positive number. '
+                             'Got {}.'.format(value))
+
+    @property
+    def y_max(self):
+        return self._y_max
+
+    @y_max.setter
+    def y_max(self, value):
+        if value is None or (isinstance(value, (float, int)) and value > 0):
+            self._y_max = value
+        else:
+            raise ValueError('y_max is expected to be a positive number '
+                             'or None. Got {}.'.format(value))
+
+    @property
+    def x_max(self):
+        return self._x_max
+
+    @x_max.setter
+    def x_max(self, value):
+        if value is None or (isinstance(value, (float, int)) and value > 0):
+            self._x_max = value
+        else:
+            raise ValueError('x_max is expected to be a positive number '
+                             'or None. Got {}.'.format(value))
+
+    @property
+    def y_min(self):
+        return self._y_min
+
+    @y_min.setter
+    def y_min(self, value):
+        if isinstance(value, (float, int)) and value >= 0:
+            self._y_min = value
+        else:
+            raise ValueError('y_min is expected to be a nonnegative number. '
+                             'Got {}.'.format(value))
+
+    @property
+    def x_min(self):
+        return self._x_min
+
+    @x_min.setter
+    def x_min(self, value):
+        if isinstance(value, (float, int)) and value >= 0:
+            self._x_min = value
+        else:
+            raise ValueError('x_min is expected to be a nonnegative number. '
+                             'Got {}.'.format(value))
+
+    def __call__(self, data, radii, mode='none'):
+        return self.evaluate(data=data, radii=radii, mode=mode)
+
+    def _pairwise_diffs(self, data):
+        npts = len(data)
+        diff = np.zeros(shape=(npts * (npts - 1) // 2, 2), dtype=np.double)
+        k = 0
+        for i in range(npts - 1):
+            for j in range(i + 1, npts):
+                diff[k] = abs(data[i] - data[j])
+                k = k + 1
+
+        return diff
+
+    def _pairwise_distances(self, data):
+        npts = len(data)
+        distances = np.zeros((npts * (npts - 1)) // 2, dtype=np.double)
+        diff = self._pairwise_diffs(data)
+        for k in range(len(distances)):
+            distances[k] = math.sqrt((diff[k] * diff[k]).sum())
+
+        return distances
+
+
+    def poisson(self, radii):
+        """
+        Evaluates the Ripley K function for the homongeneous Poisson process,
+        also known as Complete State of Randomness (CSR).
+
+        Parameters
+        ----------
+        radii : 1D array
+            Set of distances in which Ripley's K function will be evaluated.
+
+        Returns
+        -------
+        output : 1D array
+            Ripley's K function evaluated at ``radii``.
+        """
+
+        return np.pi * radii * radii
```

Comment:
I know this formula under the name of "area of a circle", when did they change that to "poisson"? 😄 Just joking.